indexd/src/lib.rs

```diff
@@ -177,13 +177,14 @@ impl SDK<ConnectedState> {
         &self,
         reader: R,
         encryption_key: EncryptionKey,
+        nonce_prefix: [u8; 16],
```

Comment:
My preference would be to not pass this in and just generate it randomly right before we need it. Then return it to the user alongside the other results. There is no need for a user to bother with this.

Not sure how @n8mgr and @alexfreska feel about this but I think it would be nice API design to have a dedicated `UploadMeta` response type for the SDK which includes the object, the encryption key as well as the nonce. 
Then all we need a user to do is persist that exact metadata.